src/record/record.ts

```diff
@@ -96,31 +96,26 @@ export class Record extends Emitter  {
     public subscribe (path: string, callback: (data: any) => void, triggerNow?: boolean) {
         const parameters = utils.normalizeArguments(arguments)
         this.subscriptions.push(parameters)
-        this.record.subscribe(parameters)
+        this.record.subscribe(parameters, this)
     }
 
     public unsubscribe (path: string, callback: (data: any) => void): void {
         const parameters = utils.normalizeArguments(arguments)
-        for (const item of this.record.references) {
-          item.subscriptions = item.subscriptions.filter((subscription: any) => {
-            if (!parameters.callback && (subscription.path === parameters.path)) return false
-            if (parameters.callback && (subscription.path === parameters.path && subscription.callback === parameters.callback)) return false
-            if (parameters.callback && subscription.callback === parameters.callback) return false
-            return true
-          })
-        }
+        this.subscriptions = this.subscriptions.filter((subscription: any) => {
+          if (!parameters.callback && (subscription.path === parameters.path)) return false
```

Comment:
i know this is older code, but when would we not have a callback? I feel like that might be broken code

Reply:
when you do `unsubscribe('path')`